src/tests/components/Grid.test.tsx

```diff
@@ -0,0 +1,32 @@
+import { render, screen } from '@testing-library/react'
+import Grid from '../../components/Grid'
+import wrapWithReduxProvider from '../utils/reduxProviderWrapper'
+import userEvent from '@testing-library/user-event'
+
+describe('components/Grid.tsx', () => {
+  it('displays a grid with desired dimensions', () => {
+    render(wrapWithReduxProvider(<Grid width={20} height={10} />))
+
+    expect(screen.getAllByRole('row')).toHaveLength(10)
+    expect(screen.getAllByRole('gridcell')).toHaveLength(200)
+  })
+
+  it('displays an initialised grid with only dead cells', () => {
+    render(wrapWithReduxProvider(<Grid width={20} height={10} />))
+
+    screen.getAllByRole('gridcell').forEach((cell) => {
+      expect(cell).toHaveStyle('background-color:grey')
+    })
+  })
+
+  it('toggles isAlive cells porperty onclick', async () => {
+    render(wrapWithReduxProvider(<Grid width={20} height={10} />))
+
+    const cell = screen.getAllByRole('gridcell')[0]
+    expect(cell).toHaveStyle('background-color:grey')
+    await userEvent.click(cell)
+    expect(cell).toHaveStyle('background-color:white')
+    await userEvent.click(cell)
+    expect(cell).toHaveStyle('background-color:grey')
+  })
```

Comment:
Je ne suis toujours pas convaincu de la pertinence de ce test. Je comprend ton argument de vouloir définir notre etat de depart nous même en cliquant sur la grille, mais c'est problématique car tu introduit du fonctionnel qui n'est pas demandé dans les specs de ton projet. 

C'est autant de code qui peut etre source de problèmes et qui a peu de chance d'etre maintenu, vu que ce qu'il fait n'a pas été demandé au départ. 

A mon avis ce composant se limite à définir une grille de la taille que tu souhaites et a la remplire avec des cellules qui ont chacune un etat de depart. 

Verifier que le comportement de tes cellules change a chaque iteration se fera dans un deuxième temps, et pas nécéssairement dans ce composant.

Reply:
Je ne sais pas trop car rien dans les specs ne décrit la manière d'initialiser le contenu de la grille. Par ailleur comme il est mentionné des patternes à essayer (vaisseaux, canon, grenouille ...), je me suis dis qu'il serais bien de pouvoir nous même préparer le contenu de la grille en cliquant sur les cellules afin de pouvoir dessiner ces patternes.

Reply:
Ok, ca s'entend :+1: Fait quand meme attention au nom de ton test, et a séparer les tests. Essayes quelque chose du genre
`it('kills a cell when a user clicks on a living one')` et `it('revive a cell when a user clicks on a dead one')`